src/main/java/uk/gov/hmcts/reform/unspec/service/documentmanagement/DocumentManagementService.java

```diff
@@ -0,0 +1,151 @@
+            MultipartFile file
+                = new InMemoryMultipartFile(FILES_NAME, originalFileName, PDF.CONTENT_TYPE, pdf.getBytes());
+
+            UserInfo userInfo = userService.getUserInfo(authorisation);
+            UploadResponse response = documentUploadClientApi.upload(
+                authorisation,
+                authTokenGenerator.generate(),
+                userInfo.getUid(),
+                userRoles,
+                Classification.RESTRICTED,
+                singletonList(file)
+            );
+
+            Document document = response.getEmbedded().getDocuments().stream()
+                .findFirst()
+                .orElseThrow(() -> {
+                    String message = "Document management failed uploading file" + originalFileName;
+                    return new DocumentManagementException(message);
+                });
+
+            return CaseDocument.builder()
+                .documentLink(uk.gov.hmcts.reform.unspec.model.documents.Document.builder()
+                                  .documentUrl(document.links.self.href)
+                                  .documentBinaryUrl(document.links.binary.href)
+                                  .documentFileName(originalFileName)
+                                  .build())
+                .documentName(originalFileName)
+                .documentType(pdf.getDocumentType())
+                .createdDatetime(LocalDateTimeHelper.nowInUTC())
+                .documentSize(document.size)
+                .createdBy(UNSPEC)
+                .build();
+        } catch (Exception ex) {
+            throw new DocumentManagementException(String.format(
+                "Unable to upload document %s to document management.",
+                originalFileName
+            ), ex);
+        }
+    }
+
+    @Retryable(value = DocumentManagementException.class, backoff = @Backoff(delay = 200))
+    public byte[] downloadDocument(String authorisation, URI documentManagementUrl) {
+        try {
+            UserInfo userInfo = userService.getUserInfo(authorisation);
+            String userRoles = String.join(",", this.userRoles);
+            Document documentMetadata = getDocumentMetaData(authorisation, documentManagementUrl.getPath());
+
+            ResponseEntity<Resource> responseEntity = documentDownloadClientApi.downloadBinary(
+                authorisation,
+                authTokenGenerator.generate(),
+                userRoles,
+                userInfo.getUid(),
+                URI.create(documentMetadata.links.binary.href).getPath()
+            );
+
+            ByteArrayResource resource = (ByteArrayResource) responseEntity.getBody();
+            //noinspection ConstantConditions let the NPE be thrown
+            return resource.getByteArray();
+        } catch (Exception ex) {
+            throw new DocumentManagementException(
+                String.format(
+                    "Unable to download document %s from document management.",
```

Comment:
How about having 2 exceptions for it, like `DocumentDownloadException` and DocumentUploadException`, thoughts?